### PR TITLE
Mobility screening

### PR DESCRIPTION
**Pull request by alexnguyen0512**
Docstring was added as a description of this builder.
Import statements were added to the file to cover dependencies. MPRester and MP2020Compatibiity objects were initialized in init statement.
Formatting was checked by black.
A test file was also added to make sure the builder works as expected.

### CODE REVIEW

File: emmet-builders/emmet/builders/mobility_screening/migration_graph.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        angle_tol: angle tolerance for StructureMatcher
+        """
+        self.electrodes = electrodes
+        self.tasks = tasks
+        self.ltol = ltol
+        self.stol = stol
+        self.angle_tol = angle_tol
+        self.migration_graphs = migration_graphs
+        self.wi_entries = {}
+        from pymatgen.ext.matproj import MPRester
```
Comment (acrutt) [inline emmet-builders/emmet/builders/mobility_screening/migration_graph.py]: The import statements for MPRester and MaterialsProject2020Compatibility should go at the top of the file. We typically do not put import statements within the builder, just at the start of the file.

File: emmet-builders/emmet/builders/mobility_screening/migration_graph.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        super().__init__(source=electrodes, target=migration_graphs, **kwargs)
+        self.sources.append(tasks)
+
+    def get_items(self) -> dict:
+        for item in super(MigrationGraphBuilder, self).get_items():
+            tds = list(self.tasks.query({"task_id": {"$in": item["material_ids"]}}))
+            item.update({"task_docs": tds})
+            yield item
+
+    # get structure entry from task store and attach migration graph dict to item
```
Comment (acrutt) [inline emmet-builders/emmet/builders/mobility_screening/migration_graph.py]: This comment seems misplaced? Should it go before line 51 or 50? The usual convention I've seen is putting descriptive comments before code or in-line if it is a short comment.

**Review by acrutt — COMMENTED** [review]
Thank you for the latest round of updates, Alex! Just a couple minor formatting fixes and then this will be ready to merge.

Also, the test file you added for now is fine and looks good as a local file for making sure the builder works as expected! However, there are conventions for writing unit tests although they are not implemented in emmet. For example, see this portion of [maggma](https://github.com/materialsproject/maggma/tree/main/tests) which uses pytest. Specifically the [tests for copy builder](https://github.com/materialsproject/maggma/blob/main/tests/builders/test_copy_builder.py) is a good simple example of how to approach writing a test file for a builder which can be run independently (it can be run by anyone because you do not need nersc database credentials). Writing a test for this builder is beyond the scope of what is needed for now so I will likely remove this test file in the future.

**Comment by alexnguyen0512** [comment]
Closed to address comment

**Comment by acrutt** [comment]
I think I initially misunderstood the intent of the comments you added. If you are writing a comment to describe what a function does, it's best to add that as a docstring with this format:

    def your_function():
        """
        your comments
        """
That way it is more accessible, like if you press shift+tab in a jupyter notebook. Here is one example for `process items` in the electrodes builder: https://github.com/materialsproject/emmet/blob/main/emmet-builders/emmet/builders/materials/electrodes.py#L405

This is some code I've written in the past that demonstrates what I was describing previously of commenting your code (with descriptive comments before code) to clarify what is happening.
https://github.com/hackingmaterials/atomate/blob/main/atomate/vasp/firetasks/approx_neb_tasks.py#L49

**Pull request closed without merging**